tests/php/ORM/DBMoneyTest.php

```diff
@@ -339,6 +340,84 @@ public function testHasAmount()
         $this->assertFalse($obj->MyMoney->hasAmount());
     }
 
+    public static function provideSetGetValue(): array
+    {
+        return [
+            'int' => [
+                'amount' => 3,
+                'currency' => 'NZD',
+                'expected' => '3 NZD',
+            ],
+            'string-int' => [
+                'amount' => '3',
+                'currency' => 'NZD',
+                'expected' => '3 NZD',
+            ],
+            'negative-int' => [
+                'amount' => -3,
+                'currency' => 'NZD',
+                'expected' => '-3 NZD',
+            ],
+            'negative-string-int' => [
+                'amount' => '-3',
+                'currency' => 'NZD',
+                'expected' => '-3 NZD',
+            ],
+            'float' => [
+                'amount' => 3.65,
+                'currency' => 'NZD',
+                'expected' => '3.65 NZD',
+            ],
+            'string-float' => [
+                'amount' => '3.65',
+                'currency' => 'NZD',
+                'expected' => '3.65 NZD',
+            ],
+            'negative-float' => [
+                'amount' => -3.65,
+                'currency' => 'NZD',
+                'expected' => '-3.65 NZD',
+            ],
+            'negative-string-float' => [
+                'amount' => '-3.65',
+                'currency' => 'NZD',
+                'expected' => '-3.65 NZD',
+            ],
+            'no-curency' => [
+                'amount' => '1.23',
+                'currency' => null,
+                'expected' => '1.23',
+            ],
+            'non-numeric-no-currency' => [
+                'amount' => 'fish',
+                'currency' => null,
+                'expected' => null,
+            ],
+            'string' => [
+                'amount' => 'fish',
+                'currency' => 'NZD',
+                'expected' => null,
```

Comment:
This we expect to return `null`, but `DBBoolean` we expect to return `"fish"` - we should be consistent. Presumably this should return `"fish NZD"` but still fail validation?

Reply:
Makes sense, have updated